### PR TITLE
Feature/conditionally disabled test execution button

### PR DESCRIPTION
**Pull request by jjelschen**

### CODE REVIEW

File: src/lib/src/component/navigation/navigation.component.spec.ts [hunk 1]
```diff
@@ -33,13 +33,20 @@ describe('NavigationComponent', () => {
   let spy: jasmine.Spy;
   let sidenav: DebugElement;
 
-  let listedFile: WorkspaceElement = {
```
Comment (gunther-bachmann) [inline src/lib/src/component/navigation/navigation.component.spec.ts:36]: Just for the future, renaming would be better suited for a separate pull. Makes reviewing much easier

File: src/lib/src/component/navigation/navigation.component.css [hunk 2]
```diff
@@ -54,6 +54,16 @@
   cursor: pointer;
 }
 
-.icon:hover {
+.icon:hover:enabled {
   color: #ffffff;
 }
+
+:disabled {
```
Comment (gunther-bachmann) [inline src/lib/src/component/navigation/navigation.component.css]: Is this for disabled elements? Is that wanted?

Comment (franzbecker) [reply src/lib/src/component/navigation/navigation.component.css]: Shouldn't it be `.icon:disabled` ?

File: src/lib/src/component/navigation/navigation.component.spec.ts [hunk 3]
```diff
@@ -364,18 +371,61 @@ describe('NavigationComponent', () => {
...
+
+  it('enables the run button when selecting an executable file', async(() => {
+    // given
+    component.workspace = new Workspace({
+      name: 'root',
+      path: '',
+      type: ElementType.Folder,
+      children: [ nonExecutableFile, tclFile ]
+    });
+    fixture.detectChanges();
```
Comment (franzbecker) [inline src/lib/src/component/navigation/navigation.component.spec.ts]: This code and the one above is redundant. We should probably extract this into a separate method.

File: src/lib/src/component/navigation/navigation.component.spec.ts [hunk 4]
```diff
@@ -364,18 +371,61 @@ describe('NavigationComponent', () => {
...
+    let runIcon = sidenav.query(By.css('#run'));
+    component.uiState.selectedElement = nonExecutableFile;
```
Comment (franzbecker) [inline src/lib/src/component/navigation/navigation.component.spec.ts]: I think this line is not doing anything. But I'm missing a test-case which is that the run button is disabled by default.

**Review by franzbecker — CHANGES_REQUESTED** [review]

File: src/lib/src/component/navigation/navigation.component.test.setup.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+export function mockedPersistenceService() {
+  // Mock PersistenceService
+  const persistenceService = mock(PersistenceService);
+  when(persistenceService.listFiles()).thenReturn(Promise.resolve(tclFile));
+  return persistenceService;
+}
+
+export function mockedTestExecutionService() {
+  // Mock TestExecutionService
```
Comment (franzbecker) [inline src/lib/src/component/navigation/navigation.component.test.setup.ts]: redundant comment

File: src/lib/src/component/navigation/navigation.component.test.setup.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+export const nonExecutableFile: WorkspaceElement = {
+  name: 'nonExecutable.txt',
+  path: 'path/to/nonExecutable.txt',
+  type: ElementType.File,
+  children: []
+};
+
+export function mockedPersistenceService() {
+  // Mock PersistenceService
```
Comment (franzbecker) [inline src/lib/src/component/navigation/navigation.component.test.setup.ts]: redundant comment

**Review by franzbecker — APPROVED** [review]

**Pull request merged**